packages/components/src/popover/index.js

```diff
@@ -42,405 +49,140 @@ const SLOT_NAME = 'Popover';
-
-		return offsetIframe(
-			new window.DOMRect(
-				topRect.left,
-				topRect.top,
-				topRect.width,
-				bottomRect.bottom - topRect.top
-			),
-			top.ownerDocument,
-			container
-		);
-	}
-
-	if ( ! anchorRefFallback.current ) {
-		return;
+		return x + suffix;
 	}
 
-	const { parentNode } = anchorRefFallback.current;
-
-	return offsetIframe(
-		parentNode.getBoundingClientRect(),
-		parentNode.ownerDocument,
-		container
-	);
-}
+	return y;
+};
 
-/**
- * Sets or removes an element attribute.
- *
- * @param {Element} element The element to modify.
- * @param {string}  name    The attribute name to set or remove.
- * @param {?string} value   The value to set. A falsy value will remove the
- *                          attribute.
- */
-function setAttribute( element, name, value ) {
-	if ( ! value ) {
-		if ( element.hasAttribute( name ) ) {
-			element.removeAttribute( name );
+const placementToAnimationOrigin = ( placement ) => {
+	const [ a, b ] = placement.split( '-' );
+
+	let x, y;
+	if ( a === 'top' || a === 'bottom' ) {
+		x = a === 'top' ? 'bottom' : 'top';
+		y = 'middle';
+		if ( b === 'start' ) {
+			y = 'left';
+		} else if ( b === 'end' ) {
+			y = 'right';
 		}
-	} else if ( element.getAttribute( name ) !== value ) {
-		element.setAttribute( name, value );
 	}
-}
 
-/**
- * Sets or removes an element style property.
- *
- * @param {Element} element  The element to modify.
- * @param {string}  property The property to set or remove.
- * @param {?string} value    The value to set. A falsy value will remove the
- *                           property.
- */
-function setStyle( element, property, value = '' ) {
-	if ( element.style[ property ] !== value ) {
-		element.style[ property ] = value;
-	}
-}
-
-/**
- * Sets or removes an element class.
- *
- * @param {Element} element The element to modify.
- * @param {string}  name    The class to set or remove.
- * @param {boolean} toggle  True to set the class, false to remove.
- */
-function setClass( element, name, toggle ) {
-	if ( toggle ) {
-		if ( ! element.classList.contains( name ) ) {
-			element.classList.add( name );
+	if ( a === 'left' || a === 'right' ) {
+		x = 'center';
+		y = a === 'left' ? 'right' : 'left';
+		if ( b === 'start' ) {
+			x = 'top';
+		} else if ( b === 'end' ) {
+			x = 'bottom';
 		}
-	} else if ( element.classList.contains( name ) ) {
-		element.classList.remove( name );
-	}
-}
-
-function getAnchorDocument( anchor ) {
-	if ( ! anchor ) {
-		return;
-	}
-
-	if ( anchor.endContainer ) {
-		return anchor.endContainer.ownerDocument;
 	}
 
-	if ( anchor.top ) {
-		return anchor.top.ownerDocument;
-	}
-
-	return anchor.ownerDocument;
-}
+	return x + ' ' + y;
+};
 
 const Popover = (
 	{
+		// Disable reason: We generate the `...contentProps` rest as remainder
+		// of props which aren't explicitly handled by this component.
+		/* eslint-disable no-unused-vars */
+		range,
+		/* eslint-enable no-unused-vars */
```

Comment:
Nit

```suggestion
		/* eslint-disable-next-line no-unused-vars */
		range,
```

Reply:
I believe it's a prop that existed a long time ago in Popover and if we drop it from there, it creates a React warning potentially for folks still using it. I'm not sure how important is it to avoid that warning though.

Reply:
We could add a soft deprecation for now, and remove in WordPress 6.2 or something?

Reply:
Makes sense yes.